nixos/modules/services/misc/languagetool.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, options, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.languagetool;
+in {
+  options.services.languagetool = {
+    enable = mkOption {
+      type = types.bool;
+      example = true;
+      default = false;
+      description = mdDoc ''
+        Start the HTTP LanguageTool server.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8081;
+      example = 8081;
+      description = mdDoc ''
+        Port on which LanguageTool listens.
+      '';
+    };
+
+    public = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = mdDoc ''
+        If `true` allow the server to be connected from anywhere,
+        otherwise only from the computer it was started from.
+      '';
+    };
+
+    allowOrigin = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://my-website.org";
+      description = mdDoc ''
+        Set the Access-Control-Allow-Origin header in the HTTP response,
+        used for direct (non-proxy) JavaScript-based access from browsers.
+        `null` to allow access from all sites.
+      '';
+    };
+
+    configuration = mkOption {
+      type = types.str;
+      default = "";
+      example = ''
+        cacheSize=1000
+      '';
+      description = mdDoc ''
+        LanguageTool configuration file content.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.etc."languagetool.conf".text = cfg.configuration;
```

Comment:
this should only be done if the file has to exist in this location (eg for interoperation with a CLI). it's usually better to write the configuration to a file with other means (eg a `pkgs.formats` generator) and use the path that generated instead of a static path in `/etc`

Reply:
OK